### PR TITLE
conn: fix lost connection when insert from select panic cause by out of memory quota

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
Set `oom-action = "cancel"` in the tidb config file.

```sql
drop table if exists t,t1;
create table t (a bigint);
create table t1 (a bigint);
insert into t1 values (1),(2),(3),(4),(5);
set @@tidb_mem_quota_query=200;
insert into t select a from t1 order by a desc;
ERROR 2013 (HY000): Lost connection to MySQL server during query
```

When `insert ... select` out of memory quota, TiDB should return error `Out Of Memory Quota`, instead of lost connection.

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

Code changes

 - Has exported function/method change

Side effects

Related changes

 - Need to cherry-pick to the release branch

Release note

 - Fix the issue that lost connection when executing `insert from select` statement panic caused by out of memory quota.

### CODE REVIEW

**Comment by crazycs520** [comment]
@zz-jason @tiancaiamao PTAL

File: session/session.go [hunk 1]
```diff
@@ -1036,6 +1038,21 @@ func (s *session) executeStatement(ctx context.Context, connID uint64, stmtNode
...
+			return
+		}
+		if str, ok := r.(string); !ok || !strings.HasPrefix(str, memory.PanicMemoryExceed) {
+			panic(r)
+		}
+		err = errors.Errorf("%v", r)
+		buf := make([]byte, 4096)
+		stackSize := runtime.Stack(buf, false)
+		buf = buf[:stackSize]
+		logutil.Logger(ctx).Error("execute sql panic", zap.String("sql", sql), zap.String("stack", string(buf)))
```
Comment (lonng) [inline session/session.go]: Prefer using `zap.Stack()` directly.

Comment (crazycs520) [reply session/session.go]: Great, done.

File: session/session.go [hunk 2]
```diff
@@ -1036,6 +1037,18 @@ func (s *session) executeStatement(ctx context.Context, connID uint64, stmtNode
 }
 
 func (s *session) Execute(ctx context.Context, sql string) (recordSets []sqlexec.RecordSet, err error) {
+	defer func() {
+		r := recover()
+		if r == nil {
+			return
+		}
+		if str, ok := r.(string); !ok || !strings.HasPrefix(str, memory.PanicMemoryExceed) {
```
Comment (zz-jason) [inline session/session.go]: I think a better place is `ExecStmt.Exec()` function, or `ExecStmt.handleNoDelay()` and `ExecStmt.handlePessimisticSelectForUpdate()` function.

Comment (crazycs520) [reply session/session.go]: done.

**Pull request merged**